tests/unit/cypher_main_visitor.cpp

```diff
@@ -3772,6 +3804,131 @@ TEST_P(CypherMainVisitorTest, CreateStream) {
+                                     const std::string_view transform_name,
+                                     const std::optional<TypedValue> &batch_interval,
+                                     const std::optional<TypedValue> &batch_size, const std::string_view service_url) {
+  SCOPED_TRACE(query_string);
+
+  StreamQuery *parsed_query{nullptr};
+  ASSERT_NO_THROW(parsed_query = dynamic_cast<StreamQuery *>(ast_generator.ParseQuery(query_string))) << query_string;
+  ASSERT_NE(parsed_query, nullptr);
+  EXPECT_EQ(parsed_query->stream_name_, stream_name);
+  ValidateTopicNames(parsed_query->topic_names_, topic_names, ast_generator);
+  EXPECT_EQ(parsed_query->transform_name_, transform_name);
+  EXPECT_NO_FATAL_FAILURE(CheckOptionalExpression(ast_generator, parsed_query->batch_interval_, batch_interval));
+  EXPECT_NO_FATAL_FAILURE(CheckOptionalExpression(ast_generator, parsed_query->batch_size_, batch_size));
+  EXPECT_EQ(parsed_query->batch_limit_, nullptr);
+  if (service_url.empty()) {
+    EXPECT_EQ(parsed_query->service_url_, nullptr);
+    return;
+  }
+  EXPECT_NE(parsed_query->service_url_, nullptr);
+}
+
+TEST_P(CypherMainVisitorTest, CreatePulsarStream) {
+  auto &ast_generator = *GetParam();
+
+  TestInvalidQuery("CREATE PULSAR STREAM", ast_generator);
+  TestInvalidQuery<SemanticException>("CREATE PULSAR STREAM stream", ast_generator);
+  TestInvalidQuery("CREATE PULSAR STREAM stream TOPICS", ast_generator);
+  TestInvalidQuery<SemanticException>("CREATE PULSAR STREAM stream TOPICS topic_name", ast_generator);
+  TestInvalidQuery("CREATE PULSAR STREAM stream TOPICS topic_name TRANSFORM", ast_generator);
+  TestInvalidQuery("CREATE PULSAR STREAM stream TOPICS topic_name TRANSFORM transform.name SERVICE_URL", ast_generator);
+  TestInvalidQuery<SemanticException>(
+      "CREATE PULSAR STREAM stream TOPICS topic_name TRANSFORM transform.name SERVICE_URL 1", ast_generator);
+  TestInvalidQuery(
+      "CREATE PULSAR STREAM stream TOPICS topic_name TRANSFORM transform.name BOOTSTRAP_SERVERS 'bootstrap'",
+      ast_generator);
+  TestInvalidQuery<SemanticException>(
+      "CREATE PULSAR STREAM stream TOPICS topic_name TRANSFORM transform.name SERVICE_URL 'test' TOPICS topic_name",
+      ast_generator);
+  TestInvalidQuery<SemanticException>(
+      "CREATE PULSAR STREAM stream TRANSFORM transform.name TOPICS topic_name TRANSFORM transform.name SERVICE_URL "
+      "'test'",
+      ast_generator);
+  TestInvalidQuery<SemanticException>(
+      "CREATE PULSAR STREAM stream TRANSFORM transform.name BATCH_INTERVAL 1 TOPICS topic_name TRANSFORM "
+      "transform.name SERVICE_URL 'test' BATCH_INTERVAL 1000",
+      ast_generator);
+  TestInvalidQuery<SemanticException>(
+      "CREATE PULSAR STREAM stream TRANSFORM transform.name BATCH_INTERVAL 'a' TOPICS topic_name TRANSFORM "
+      "transform.name SERVICE_URL 'test'",
+      ast_generator);
```

Comment:
```suggestion
  TestInvalidQuery<SemanticException>(
      "CREATE PULSAR STREAM stream TRANSFORM transform.name BATCH_INTERVAL 'a' TOPICS topic_name TRANSFORM "
      "transform.name SERVICE_URL 'test'",
      ast_generator);
```
For this I would definitely remove the second `TRANSFORM` because if I am not mistaken, it meant to check the invalid `BATCH_INTERVAL`.